lib/bademagic_module/bluetooth/scanstate.dart

```diff
@@ -0,0 +1,62 @@
+import 'dart:async';
+import 'package:badgemagic/bademagic_module/bluetooth/connectstate.dart';
+import 'package:flutter_blue_plus/flutter_blue_plus.dart';
+import 'package:logger/logger.dart';
+import 'ble_state_interface.dart';
+import 'bletoast.dart';
+
+class ScanState implements BleState {
+  ScanResult? foundDevice;
+  final Logger logger = Logger();
+  BleStateToast toast = BleStateToast();
+
+  @override
+  Future<BleState?> isFailed(String message) async {
+    toast.failureToast(message);
+    return null;
+  }
+
+  @override
+  Future<BleState?> isSuccess(String message) async {
+    toast.successToast(message);
+    return ConnectState(scanResult: foundDevice!).processState();
+  }
+
+  @override
+  Future<BleState?> processState() async {
```

Comment:
You're not making use of the returned BLEState. Every state returns null. I'd advise cleaning this up and returning the next state rather than making nested calls. Letting the main thread run until completion